app/src/main/java/com/example/playlistmaker/search/ui/SearchHistory.kt

```diff
@@ -11,7 +11,13 @@ class SearchHistory(private val localStorage: LocalStorage) {
     }
 
     fun add(track: Track) {
-        tracksList.remove(track)
+        var index = tracksList.size - 1
+        while (index >= 0) {
```

Comment:
🍏 Можно упростить всю конструкцию, используя библиотечный метод:
```suggestion
        tracksList.removeAll { it.id == track.id }
```